### ISSUE TITLE
[Narrator + New Edge] [NVDA + Chrome ] DetailList Navigation to row and columns blocked

### ISSUE THREAD

**Issue by vineetrawat** [microsoft/fluentui#15072]
### Environment Information

- **Package version(s)**:  7.105.0
- **Browser and OS versions**: 
Edge: 85.0.564.51
Chrome: Version 85.0.4183.102
Windows 10 : 10.0.19041

### Describe the issue:
When using Screen Reader(, Not able to navigate to detail list columns or data rows using the arrow key. Blocks user from taking important actions.

### Please provide a reproduction of the issue in a codepen:

The problem can be replicated on the documentation page itself.

**Doc URL**: https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist
**Codepen**: https://codepen.io/pen/?&editable=true=https%3A%2F%2Fdeveloper.microsoft.com%2Fen-us%2Ffluentui

**Repro Steps**
1) Click on Filter By Name search box
2) Click Tab to navigate to headers. Use arrow keys to navigate to various columns.
3)  Click Tab to navigate to detail rows. Use arrow keys to navigate to rows.

The above will work fine in a normal scenario

4) Now open (Narrator + Edge) or (Chrome+ NVDA) and Perform the above 1 to 3 steps

#### Actual behavior:
User cannot navigate columns or row using arrow keys.

#### Expected behavior:
Should be able to navigate columns or row using arrow keys.

### Documentation describing expected behavior
Did not find any documentation that indicates the navigation with Screen readers is not supported?

**Comment by vineetrawat** [comment]
Correcting the codepen link 
https://codepen.io/vineetrawat/pen/abNabbL?editable=true%3Dhttps%3A%2F%2Fdeveloper.microsoft.com%2Fen-us%2Ffluentui

**Comment by miroslavstastny** [comment]
@vineetrawat Thanks for the report.

I am not able to reproduce any issue in Edge 85.0.564.51 with Narrator - could you please provide more details?

I see the following issues in Chrome/NVDA, @vineetrawat is that the issue you are reporting?
- navigation in the header using left and right arrow does not work as expected
- navigation in the header using up and down arrow works but there are still some unexpected items
- once navigated to the detail rows using down arrow, arrows navigate through cells
- but if I navigate to the details using Tab key, navigation between rows using up and down arrow works as expected

See the screen capture for details:
![output](https://user-images.githubusercontent.com/9615899/93513434-226f1480-f926-11ea-9a69-fb154f295ac1.gif)

FYI @smhigley, @kolaps33

**Comment by paulgildea** [comment]
@dzearing / @joschect (I thought you recently updated DetailsList not sure if it's related) Thoughts here?

**Comment by smhigley** [comment]
@vineetrawat just throwing out a guess here -- are you sure you're trying to use arrow keys when in the correct mode in Narrator and NVDA? Both work fine for me, but the behavior you're describing could be coming from using scan mode/browse mode arrow key interactions (which are handled by the screen reader). If you want to use the DetailsList keyboard navigation, you must be in forms mode/app mode.

@miroslavstastny same as above :). But also whatever you're using to capture and display key presses is really cool! Can you share what it is? 😄

**Comment by vineetrawat** [comment]
@smhigley Thanks, You are right I was in Scan mode when using the Narrator. I switched the mode and used the down arrow key to navigate. But confirm navigation happens column by column. 

I expecting that navigation could go row by row when used down and up arrow keys. Just like @miroslavstastny showed in his screenshare.

**Comment by vineetrawat** [comment]
Closing this bug as our accessibility team did not report this issue.
